### ISSUE TITLE
Swipe Gesture fires twice on UWP for Right/Left

### ISSUE THREAD

**Issue by solomonfried** [xamarin/Xamarin.Forms#4432]
[TestSwipeGesture.zip](https://github.com/xamarin/Xamarin.Forms/files/2586267/TestSwipeGesture.zip)

### Description

Swiping Left or Right (not Up or Down) causes the swipe event to fire twice.

### Steps to Reproduce
See attached example.
Create a view in a form (I used a Box View) and attach 4 Swipe recognizers, 1 for each direction.
Handle the Swipe event by incrementing a counter and displaying the count on the screen.

### Expected Behavior

The count should increase by 1 for each swipe.

### Actual Behavior

Count increases by 2 for Right/Left swipes (on UWP, works fine on Android, not tested on ioS)

### Basic Information

- Version with issue:  Xamarin.Forms 3.3.0.967583
- Last known good version:
- IDE: VS2017
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:  Windows 10.0 Build 16299
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Reproduction Link

See attached

**Comment by kingces95** [comment]
On 3.4.0 head I'm able to get one swipe per gesture.

![b88c56eb-d50b-4810-9457-ecaade468447](https://user-images.githubusercontent.com/4120386/48654849-54bf2c00-e9b4-11e8-9d44-2881c8d23f54.GIF)

**Comment by solomonfried** [comment]
I updated to Xamarin 3.4.0.1008975. Using VS2017 15.9.1

I can see your example and I can duplicate your results as long as I keep the swipes very short.
If the swipe is longer (true on a PC with a mouse or on a touch tablet using fingers), then multiple swipes are being registered in all directions.

On Android, it always records a single swipe and the swipe ends when the finger is lifted regardless of the swipe length.

[TestSwipe.zip](https://github.com/xamarin/Xamarin.Forms/files/2592884/TestSwipe.zip)



I have uploaded a video of this behavior. The source code is the same as the original.

(In my application I now ignore any swipe that occurs with 100ms of a previous swipe in order to work around this issue.
